app/src/main/java/io/homeassistant/companion/android/sensors/SensorDetailFragment.kt

```diff
@@ -127,6 +129,13 @@ class SensorDetailFragment(
                                         requestPermissions(permissions)
                                     }
                                 )
+                            } else if (sensorManager is LastAppSensorManager) {
+                                val pm = context.packageManager
+                                val appInfo = pm.getApplicationInfo(context.packageName, 0)
+                                val appOpsManager = context.getSystemService(Context.APP_OPS_SERVICE) as AppOpsManager
+                                val mode = appOpsManager.checkOpNoThrow(AppOpsManager.OPSTR_GET_USAGE_STATS, appInfo.uid, appInfo.packageName)
+                                if (mode != AppOpsManager.MODE_ALLOWED)
+                                    requestPermissions(permissions)
```

Comment:
This is duplicate code that should be turned into a common function if we really need it.